dts/common/yaml/zephyr-i2c.yaml

```diff
@@ -0,0 +1,19 @@
+---
+title: Zephyr specific I2C properties
+id: i2c
+version: 0.1
+
+description: >
+    This binding gives Zephyr specific I2C properties
+
+properties:
+  - zephyr,speed :
```

Comment:
In case we would like to keep both properties, one defining the suggested/default/current speed and one the maximum allowed speed we need to do it differently. The "clock-frequency" property defined in dts/common/yaml/i2c.yaml  is actually the suggested/default/current speed. That's the original meaning taken from Linux DTS which we, as I understand from the previous discussions, want to honor. We've changed the original description of "clock-frequency" from "Desired I2C bus clock frequency in Hz" to "Maximum I2C bus clock frequency in Hz" overloading its meaning. If we don't like the overloaded meaning we should return to its original one. And then create another property for the maximum allowed speed, if we deem it necessary. Otherwise it all becomes a bit messy.